engines/enginetest/testutils.go

```diff
@@ -114,3 +123,101 @@ func buildRunSandbox(b engines.SandboxBuilder) bool {
 	defer evalNilOrPanic(resultSet.Dispose, "Failed to dispose of ResultSet")
 	return resultSet.Success()
 }
+
+// run helper
+type run struct {
+	provider       *engineProvider
+	context        *runtime.TaskContext
+	control        *runtime.TaskContextController
+	sandboxBuilder engines.SandboxBuilder
+	sandbox        engines.Sandbox
+	resultSet      engines.ResultSet
+	logReader      io.ReadCloser
+	closedLog      bool
+}
+
+func (p *engineProvider) NewRun(engine string) *run {
+	p.ensureEngine(engine)
+	r := run{}
+	r.provider = p
+	r.context, r.control = p.newTestTaskContext()
+	return &r
+}
+
+func (r *run) NewSandboxBuilder(payload string) {
+	assert(r.sandboxBuilder == nil, "NewSandboxBuilder already called!")
+	sandboxBuilder, err := r.provider.engine.NewSandboxBuilder(engines.SandboxOptions{
+		TaskContext: r.context,
+		Payload:     parseTestPayload(r.provider.engine, payload),
+	})
+	r.sandboxBuilder = sandboxBuilder
+	nilOrpanic(err, "Error creating SandboxBuilder")
+}
+
+func (r *run) StartSandbox() {
+	assert(r.sandbox == nil, "StartSandbox already called!")
+	sandbox, err := r.sandboxBuilder.StartSandbox()
+	nilOrpanic(err, "Failed to start sandbox")
+	r.sandbox = sandbox
+}
+
+func (r *run) WaitForResult() bool {
+	assert(r.resultSet == nil, "WaitForResult already called!")
+	resultSet, err := r.sandbox.WaitForResult()
+	nilOrpanic(err, "WaitForResult failed")
+	r.resultSet = resultSet
+	r.closedLog = true
+	nilOrpanic(r.control.CloseLog(), "Failed to close log writer")
+	return r.resultSet.Success()
+}
+
+func (r *run) OpenLogReader() {
+	assert(r.logReader == nil, "OpenLogReader already called!")
+	logReader, err := r.context.NewLogReader()
+	nilOrpanic(err, "Failed to open log reader")
+	r.logReader = logReader
+}
+
+func (r *run) ReadLog() string {
+	reader, err := r.context.NewLogReader()
+	defer reader.Close()
+	nilOrpanic(err, "Failed to open log reader")
+	data, err := ioutil.ReadAll(reader)
+	nilOrpanic(err, "Failed to read log")
+	return string(data)
+}
+
+func (r *run) Dispose() {
+	if r.sandboxBuilder != nil {
+		nilOrpanic(r.sandboxBuilder.Discard(), "")
```

Comment:
@jonasfj I've addressed this one separately in #27 as it wasn't introduced in this PR.